### PR TITLE
Add ^:multipart option to defendpoint

### PR DESCRIPTION
**Pull request by tsmacdonald**
(Allows file uploads)

This will be used for CSV uploads (#29561) but I wanted to get it into master as soon as I could.

I'll add more robust tests once we have endpoints that use it—I'd rather test that file attachment works for endpoints we care about than some contrived example in `api.common-test`

Sample usage:

```clojure
 (api/defendpoint ^:multipart POST "/from-csv"
   "Create a table and model populated with the values from the attached CSV."
   [:as {raw-params :params}]
   (let [f (get raw-params "file")]
     (println (slurp (:tempfile f)))
     {:status 200}))
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29607)

### CODE REVIEW

**Comment by calherries** [comment]
You can just add `wrap-multipart-params` middleware to `metabase.server.handler/middleware`, and do without the `:multipart` metadata. The performance cost of `wrap-multipart-params` is negligible.

**Comment by tsmacdonald** [comment]
> You can just add `wrap-multipart-params` middleware to `metabase.server.handler/middleware`, and do without the `:multipart` metadata

I strongly suspect we're going to eventually pass the [`:progress-fn`](https://ring-clojure.github.io/ring/ring.middleware.multipart-params.html) in to the `wrap-multipart-params` call on a per-endpoint basis, which I expect will happen through metadata. It means that instead of a boolean we'll have to pass in a map, but we can deal with that later.

**Comment by dpsutton** [comment]
> I strongly suspect we're going to eventually pass the [:progress-fn](https://ring-clojure.github.io/ring/ring.middleware.multipart-params.html) in to the wrap-multipart-params call on a per-endpoint basis, which I expect will happen through metadata. It means that instead of a boolean we'll have to pass in a map, but we can deal with that later.

Can you explain what you are thinking here? Who would be the consumesr of the `:progress-fn`?

File: bin/build/resources/overrides.edn [hunk 1]
```diff
@@ -41,6 +41,7 @@
  "io.dropwizard.metrics"       {"metrics-core" {:resource "apache2_0.txt"}}
  "jakarta.activation"          {"jakarta.activation-api" {:resource "EDL.txt"}}
  "jakarta.xml.bind"            {"jakarta.xml.bind-api" {:resource "EDL.txt"}}
+ "javax.servlet"               {"servlet-api" {:resource "GPL2_0.txt"}}
```
[the remark this answers is not in the document]
Comment (dpsutton) [reply bin/build/resources/overrides.edn:44]: let's add the text of the cddl and use that. You can get a version at https://opensource.org/license/cddl1-txt/ and put the text in `bin/build/resources`.

**Review by dpsutton — APPROVED** [review]
approved. don't love the macroexpansion test but i don't have any better ideas. I think it's worth it to throw the cddl license in there but not super important.

**Comment by dpsutton** [comment]
@tsmacdonald one thing I noticed. You added this for `defendpoint` but not `defendpoint-schema`. I suspect this might surprise people.

**Comment by tsmacdonald** [comment]
@dpsutton I added it to `defendpoint*`, which is used by both the Plumatic and Malli defendpoints

**Comment by dpsutton** [comment]
@tsmacdonald good call. sorry i missed that.

**Pull request merged**